### ISSUE TITLE
Fan coil unit hot water coil waterside control problem

### ISSUE THREAD

**Issue by EnergyArchmage** [NREL/EnergyPlus#5276]
Helpdesk issue 10848.  User noticed a severe mismatch in heating energy between the water side and air side during some times of the day.  The water flow is sometimes running at full mass flow rate and large temperature difference and the McdeltaT can be a factor of 10 higher than the air side coil heating rate.  

Fan coil unit has many places where there is While loop that finds a PLR, need make sure that each one of those has code like this before it calls Calc4PipeFanCoil.

```
mdot = PLR * FanCoil( FanCoilNum ).MaxHotWaterFlow;

SetComponentFlowRate( mdot, FanCoil( FanCoilNum ).HotControlNode, FanCoil( FanCoilNum ).HotPlantOutletNode, FanCoil( FanCoilNum ).HWLoopNum, FanCoil( FanCoilNum ).HWLoopSide, FanCoil( FanCoilNum ).HWBranchNum, FanCoil( FanCoilNum ).HWCompNum );
```

**Comment by EnergyArchmage** [comment]
Not having any luck with this.  Initial idea did not work out.  I have tried a number of code changes and nothing seems to work.  Cannot reproduce in an example file.  

In additon to the water side showing more heat transfer that the air side, it also can show air side heat transfer with no water mass flow at all.   

Does not appear that I can deliver a fix on this, have to cut and run at this point.

**Comment by wfbuhl** [comment]
Oh this is ugly. We have a zone in a big, complicated building with only a fancoil. Some of the time in heating mode the fancoil works well - hw flow and heating output move in sync and the zone temperature is controlled. At other times the zone temperature is not well controlled. The hw flow goes from either max (zone temp below set point) to zero (zone temperature above setpoint). Meanwhile hw coil output seems to be random. The good news is that the hw flow rate seems to be trying to do the right thing, but the air side (the hw coil output) is not responding. Could there be something wrong with the hw coil model? Or how the coil output is calculated? This should be interesting.

**Comment by Myoldmopar** [comment]
OK, @mjwitte, this appears to have gone through @EnergyArchmage and @wfbuhl without much promise of a fix yet.  If we want this included in the bug fix release, I guess we need to have a call and figure out a team effort way to fix it??  Or we push it off until after the bug fix release.  What do you think?

**Comment by mjwitte** [comment]
The best thing would be to try to reproduce it with a smaller file.  Is this happening with a particular fan coil control type?

**Comment by wfbuhl** [comment]
I'm still working on it. This is a complicated bug, involving interaction
between Plant and the solver used in the fan coil. I'm trying something
right now that might work (different solver - RegulaFalsi) and messing with
FlowLock, but I'm not real happy with this even if it works.

The good news is that if you tell the user to switch to "    CyclingFan,
           !- Capacity Control Method" the problem goes away. Because even
though Plant has imposed flow lock on the how water flow rate, the fan part
load ratio can still vary to meet the required Qreq.

Whereas "     ConstantFanVariableFlow, !- Capacity Control Method" has only
one degree of freedom (the HW flow) and when this is locked it can't
respond.

The original problem is coming from Plant where

if ( AnyPlantSplitterMixerLacksContinuity() ) {
// rerun systems in a "Final flow lock/last iteration" mode
// now call for one second to last plant simulation

the splitter doesn't apparently have a mass flow balance and flow lock is
imposed. I haven't gone into this. I'm just trying to get fan coil to
respond appropriately when flow lock is imposed. The splitter has like 54
branches.

Obviously this bug is a real time suck. It may not be solved by interim
release.

Oh - I'm not optimistic about a simpler file. I've simplified the user file
by eliminating shading and going to a 3 day run period. It is not a real
barrier to debugging, though it is still kind of slow.

Fred

**Comment by Myoldmopar** [comment]
Thanks @wfbuhl for the update.  Given that there _is_ a workaround for the users to at least get them (and the solver) unstuck, I don't think this will hold up the bug fix release.  But we'll still hope.

**Comment by wfbuhl** [comment]
Whoa! The fix works! But you are going to have to look at the code (Brent
also) because it is funky. Also the solution method is changed, which will
affect all fan coils with constant fan. In other words, it might crap out
in some other example file. Still, this might be the fix. I'm going to have
to clean up the code then I will upload a new branch to GitHub.

**Comment by wfbuhl** [comment]
Also I don't have any unit tests.

**Comment by rraustad** [comment]
I'll have to merge this in with the SZVAV feature being developed now. Please address this ASAP as I also want to switch to a RegulaFalsi approach. This might be difficult to merge although I am trying to steer clear of existing code.

**Comment by wfbuhl** [comment]
Well Richard since I was dilatory I got to do the code merge. It was messier than I thought it would be but it looks like the merge is OK

**Comment by EnergyArchmage** [comment]
defect file uploaded to

https://github.com/NREL/EnergyPlusDevSupport/tree/master/DefectFiles/5000s/5276

uses ITA_Bolzano.160200_IGDG.epw

here is a plot of the original problem in water mass flow rates
![plot of problem](https://cloud.githubusercontent.com/assets/8754769/13846867/1947d434-ec18-11e5-8522-11ca98197577.png)

**Comment by rraustad** [comment]
Closed via #5539.
